code/datums/components/_component.dm

```diff
@@ -298,6 +298,15 @@
 		return list()
 	return islist(components) ? components : list(components)
 
+/**
+ * Gets all components with no typepath provided.
+ */
+/datum/proc/GetAllComponentTypes()
+	var/list/components = list()
+	for(var/component_type in _datum_components)
+		components += component_type
+	return components
```

Comment:
This seems a bit overcomplicated for what can be accomplished with `_datum_components.Copy()` or `assoc_to_keys(_datum_components)`